doc/Overview/Mvux/Advanced/Commands.md

```diff
@@ -0,0 +1,191 @@
+---
+uid: Overview.Mvux.Advanced.Commands
+---
+
+# Commands
+
+## Commands recap
+
+In the [overview](xref:Overview.Mvux.Overview) you've seen a brief introduction to how a method in the Model gets generated as a command in the Bindable Proxy Model.
+
+## Asynchronous commands
+
+The [`AsyncCommand`](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive/Presentation/Commands/AsyncCommand.cs) class, is a Command that implements [`ICommand`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand?view=windows-app-sdk-1.3) and adds support for asynchronous operations.  
+As it implements `ICommand`, it can be bound from the View as a regular Command (e.g. in a `Button.Command` property). The advantage is that it can be invoked asynchronously.  
+In addition, it also implements [`INotifyPropertyChanged`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged), to enable tracking if its properties have changed, and `ILoadable` - an Uno interface that provides information of an objects state whether it's currently in execution mode or not.
+
+![A class diagram of System.ComponentModel.ICommand inheritence structure](../Assets/Commands-2.jpg)
+
+## Creating commands
+
+There are several methods of how to create an MVUX command.
+
+1. Using code generation
+
+    ### Basic commands
+
+    When creating a method in the Model, it will be generated as an `AsyncCommand` in the Bindable Model.
+
+    For example, when declaring a method of the following signatures:
+
+    ```csharp
+    public void DoWork();
+    public ValueTask DoWork();
+    public ValueTask DoWork(CancellationToken ct);    
+    ```
+
+    A `DoWork` command will be generated in the Bindable Proxy Model. When that command is executed via a button-click etc., the method in the Model will be called:
+
+    ```xml
+    <Button Command="{Binding DoWork}" />
+    ```
+
+    As you may have noticed, the method can be asynchronous and take a `CancellationToken`, but these are not mandatory and commands will be generated for synchronous methods as well.
+
+    ### Adding command parameters
+
+    An additional parameter can be added to the method, which is then assigned with the value of the `CommandParameter` received from the View. For instance, when using a Button and clicking it, the method will be called with the [`Button.CommandParameter`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.primitives.buttonbase.commandparameter) value, given that the type of the value matches the method parameter type. Otherwise the command's `CanExecute` will be false thereby disabling the button:
+
+    ```csharp
+    public void DoWork(int param)
+    {
+        ...
+    }
+    ```
+
+    ```xml
+    <Button Command="{Binding DoWork}" CommandParameter="{Binding MyIntegerValue}" />
```

Comment:
This isn't a great example, since the method can just access the MyIntegerValue in the model. It would be better to bind to the value of another element eg a Slider